macaroons/service.go

```diff
@@ -260,3 +264,47 @@ func (svc *Service) ListMacaroonIDs(_ context.Context) ([][]byte, error) {
 
 	return rootKeySlice, nil
 }
+
+// DeleteMacaroonID removes one specific root key ID. If the root key ID is
+// found and deleted, it will be returned.
+func (svc *Service) DeleteMacaroonID(
+	_ context.Context, rootKeyID []byte) ([]byte, error) {
+
+	svc.rks.encKeyMtx.RLock()
+	defer svc.rks.encKeyMtx.RUnlock()
+
+	// Check it's unlocked.
+	if svc.rks.encKey == nil {
+		return nil, ErrStoreLocked
+	}
+
+	// Deleting encryptedKeyID or DefaultRootKeyID is not allowed.
+	if bytes.Equal(rootKeyID, encryptedKeyID) ||
```

Comment:
We should also check that the ID is not nil or empty here.

Reply:
fixed